Merging MSRE Xe Poisoning Model (#788)
idaholab/virtual_test_bed: doc/content/msr/msre/msre_xe.md

# MSRE Multiphysics Xenon Poisoning Model

*Contact: Jun Fang, fangj.at.anl.gov*

> *Model link: [MSRE Xe Poisoning Model](https://github.com/hapfang/virtual_test_bed/tree/xe/msr/msre/xe_poisoning)*

Reviewer: let's use the vtb address

> !tag name=Molten Salt Reactor Experiment SAM Model

Author: Sorry for the confusion. I forgot to examine the tag properties after copying them from an existing SAM model md file. Will double check and fix them carefully.

     description=Steady state and reactivity insertion accident models of the MSRE primary loop

>      image=https://github.com/hapfang/virtual_test_bed/blob/xe/doc/content/media/msr/msre/xe_poisoning/ss_results.png

Reviewer: same

     pairs=reactor_type:MSR
                       reactor:MSRE
                       geometry:primary_loop
                       codes_used:BlueCrab;Griffin;Pronghorn

>                        transient:steady_state;RIA

Author: No RIA is modeled here, will remove it

>                        V_and_V:validation

Author: will remove the `validation` as well since there was no direct comparison to the experimental data in the original development. Will change it to `demonstrate`.

                       input_features:checkpoint_restart
                       computing_needs:Workstation
                       fiscal_year:2024
                       sponsor:NEAMS
                       institution:ANL;INL

## Summary

> This page documents an effort that extends the existing MSRE multiphysics Griffin–Pronghorn framework by incorporating Xe poisoning physics. The objective is to integrate fission-product transport and reactivity feedback effects into the established loop-scale multiphysics model so reactor dynamics reflect both thermal-hydraulic and neutronic responses to ($^{135}\text{Xe}$) and ($^{135}\text{I}$) behavior in circulating fuel salt.

Reviewer: ```suggestion
This page documents an effort that extends the existing MSRE multiphysics Griffin–Pronghorn studies by incorporating Xe poisoning physics. The objective is to integrate fission-product transport and reactivity feedback effects into the established loop-scale multiphysics model so reactor dynamics reflect both thermal-hydraulic and neutronic responses to ($^{135}\text{Xe}$) and ($^{135}\text{I}$) behavior in circulating fuel salt.
```
or model

Reviewer: TH response to Xe seems weird to me. I'd mention neutronic first, and something like "downstream effects on TH"

Reviewer: not addressed

> Modeling xenon effects is essential because ($^{135}\text{Xe}$), with its large neutron absorption cross section, has a dominant influence on reactor reactivity, stability, and operational flexibility. In molten salt reactors, this challenge is further complicated by the flowing fuel, which transports xenon throughout the primary loop and introduces additional physics such as advection, diffusion, bubble stripping, and graphite interactions. These coupled phenomena can significantly alter spatial xenon distributions and transient reactivity behavior, directly impacting reactor control, restart capability, and load-following performance.

Reviewer: you're not modeling any of these other physics here.

please add a note to include what physics are present

Reviewer: actually you do have bubble stripping. however this is the least intuitive name for this.
Let's refer to it as a gaseous fission product removal

\begin{equation}
\frac{\partial \epsilon C_i}{\partial t} +
\nabla \cdot (\vec{u} C_i) -
\nabla \cdot \left( \epsilon \left( D_{m,i} + \frac{\nu_t}{Sc_t} \right) \nabla C_i \right) = S_i - \lambda_i \epsilon C_i
\end{equation}

where $\epsilon$ is porosity (salt volume fraction), $\vec{u}$ is superficial velocity, $D_{m,i}$ is molecular diffusion coefficient, $\nu_t$ is turbulent eddy viscosity, $Sc_t$ is turbulent Schmidt number, $S_i$ is source term, and $\lambda_i$ is decay constant.

For the xenon-iodine chain, the source terms are:

- For $^{135}I$: $S_I = \gamma_I \Sigma_f \phi$
- For $^{135}Xe$: $S_{Xe} = \gamma_{Xe} \Sigma_f \phi + \lambda_I C_I$

#### Thermal Hydraulics (Pronghorn)

Pronghorn uses a weakly compressible porous-media formulation. Momentum conservation follows Darcy-Forchheimer behavior:

\begin{equation}
\frac{\rho}{\epsilon} \left( \frac{\partial \vec{u}}{\partial t} + \frac{(\vec{u} \cdot \nabla) \vec{u}}{\epsilon} \right) = -\nabla P + \rho \vec{g} - \left( \frac{\mu}{K} + \frac{\rho C_d}{\sqrt{K}} |\vec{u}| \right) \vec{u}
\end{equation}

#### Neutronics (Griffin)

> The neutron flux is computed with a 16-group diffusion approximation:

Author: Good question. I received the cross section from Mauricio. This model was developed in FY24. I will try to dig the email communications and add more context for the cross section source.

\begin{equation}
-\nabla \cdot D_g \nabla \phi_g + \Sigma_{r,g} \phi_g - \sum_{g' \neq g} \Sigma_{s,g' \to g} \phi_{g'} = \frac{\chi_g}{k_{eff}} \sum_{g'} \nu \Sigma_{f,g'} \phi_{g'}
\end{equation}

#### Reactivity Feedback and Xenon Poisoning

> Coupling between physics is achieved through cross-section updates. The macroscopic absorption cross section in Griffin is updated with local xenon concentration from Pronghorn:

Author: will clarify it

#### Mesh

The simulation uses a 2-D axisymmetric (RZ) coordinate system.

- **Geometry:** The mesh represents the full primary loop, including the core, plenums, and piping regions.
- **Initialization:** `FileMeshGenerator` loads a pre-existing mesh and restart-ready state. In practice, using Exodus restart settings (e.g., `use_for_exodus_restart = true`) allows the xenon study to start from a previously converged flow field.

!listing msr/msre/xe_poisoning/th_xe.i block=Mesh language=cpp

#### Variables

This block defines the finite-volume primary variables for the coupled thermal-hydraulic and species problem:

- **Flow & energy:** Superficial velocity components ($u_x$, $u_y$), pressure ($P$), fluid temperature ($T_{fluid}$), and solid temperature ($T_{solid}$).
- **Delayed neutron precursors (DNP):** Six groups (`c1` through `c6`) that drift with circulating salt.
- **Fission-product species:** `I135` and `Xe135` as transport scalars for poisoning-chain dynamics.

!listing msr/msre/xe_poisoning/th_xe.i block=Variables language=cpp

#### Modules / NavierStokesFV

> `NavierStokesFV` streamlines porous-media thermal-hydraulic setup:

Author: Got you.

- **Physics:** Weakly compressible porous-media formulation with Darcy-Forchheimer resistance to represent pressure loss in core and piping.
- **Coupling:** `external_heat_source = 'power_density'` links TH energy deposition to Griffin-computed neutronics power.
- **Numerics:** Upwind advection interpolation supports robust transport in high-velocity loop regions.

!listing msr/msre/xe_poisoning/th_xe.i block=Modules language=cpp

#### FVBCs

Boundary conditions for species transport primarily enforce symmetry:

- Symmetry scalar BCs on centerline/top boundaries for `I135` and `Xe135` preserve the 2-D axisymmetric assumption and prevent non-physical cross-axis species flux.

!listing msr/msre/xe_poisoning/th_xe.i block=FVBCs language=cpp

> #### FVKernels

Author: Will do.

This section contains specific source, reaction, and component-level kernels:

- **Source & Decay:** Handles fission production and radioactive decay for I-135, Xe-135, and the six DNP groups.
- **Pump & Heat Transfer:** Includes `INSFVPump` kernels for momentum and `NSFVEnergyAmbientConvection` for heat exchange in the pump/HX block.
- **Stripping Model:** A dedicated `NSFVEnergyAmbientConvection` kernel in the `strip` block represents physical xenon removal from the salt.

!listing msr/msre/xe_poisoning/th_xe.i block=FVKernels language=cpp

#### FVInterfaceKernels

Conjugate heat transfer (CHT) across fluid/structure interfaces is handled here:

- Interface kernels (e.g., convection-correlation based) compute local heat transfer between fuel salt and structures such as the core barrel, ensuring realistic thermal coupling.

!listing msr/msre/xe_poisoning/th_xe.i block=FVInterfaceKernels language=cpp

#### Materials

Material models provide local properties and closures:

- **Porosity:** Core porosity set to 22.28% to represent effective salt-channel volume fraction.

> - **Fluid properties:** Temperature-dependent (and pressure-dependent when needed) density, viscosity, and thermal conductivity.

Reviewer: pressure dependent in the salt?

- Griffin (`neu_xe.i`) is launched as a sub-application and typically executed at the end of each Pronghorn timestep/Picard update.

!listing msr/msre/xe_poisoning/th_xe.i block=MultiApps language=cpp

#### Transfers

Transfers implement two-way coupling data exchange:

- **To Griffin:** Pronghorn sends thermal state (e.g., `T_salt`), flow-related fields, DNP groups, and `Xe135` concentration.
- **From Griffin:** Pronghorn receives `power_density` and `fission_source` to update heat deposition and isotope source terms.

!listing msr/msre/xe_poisoning/th_xe.i block=Transfers language=cpp

### Griffin Input File (`neu_xe.i`)

The Griffin input file is the **Sub-Application**. It solves neutronics to determine flux, power generation, and xenon-driven reactivity effects.

#### TransportSystems

Griffin uses a 16-group CFEM-diffusion transport system:

- **Eigenvalue solve:** Computes $k_{eff}$ and flux distribution.
- **DNP linkage:** External DNP settings allow Griffin to consume precursor information transferred from Pronghorn instead of solving an independent precursor transport problem.

!listing msr/msre/xe_poisoning/neu_xe.i block=TransportSystems language=cpp

#### Mesh

> The neutronics solve uses an RZ mesh loaded via `FileMeshGenerator`, consistent with loop geometry used by the master app.

Reviewer: ```suggestion
The neutronics solve uses an RZ mesh loaded via `FileMeshGenerator`, consistent with loop geometry used by the parent application.
```

Reviewer: i think you committed these online then overrode them by force pushing

#### UserObjects

User objects support restart and solution persistence:

- Solution vectors (transport and auxiliary state) are written for restartability and postprocessing continuity.

!listing msr/msre/xe_poisoning/neu_xe.i block=UserObjects language=cpp

#### PowerDensity

This block converts fission rate into volumetric power density ($W/m^3$):

- Power is normalized to the operating reactor level (5.0 MW) so the returned thermal source is physically consistent for Pronghorn energy equations.

!listing msr/msre/xe_poisoning/neu_xe.i block=PowerDensity language=cpp

#### Materials

`CoupledFeedbackNeutronicsMaterial` is the core poisoning-feedback mechanism:

- **Cross-section feedback:** Microscopic cross-sections are interpolated versus local fuel temperature (`T_{salt}`).
- **Xenon integration:** Including `XE135` in isotope feedback and linking to `ad_Xe135` adds explicit xenon absorption penalties in the eigenvalue calculation.
- **Reactivity impact:** This coupling pathway enables prediction of xenon-induced reactivity loss (e.g., the ~-410 pcm level reported in results).

!listing msr/msre/xe_poisoning/neu_xe.i block=Materials language=cpp

## Results

> The current Pronghorn-Griffin configuration represents an initial implementation of Xe-poisoning physics in a 2-D MSRE multiphysics framework. At this stage, results are limited to steady-state behavior; transient cases and additional Xe interactions with graphite and circulating bubbles are reserved for future studies.

Reviewer: ```suggestion
The current Pronghorn-Griffin configuration represents an initial implementation of Xe-poisoning physics in a 2-D MSRE multiphysics model. At this stage, results are limited to steady-state behavior; transient cases and additional Xe interactions with graphite and circulating bubbles are reserved for future studies.
```
this isnt a framework. the tools are being used as is

Reviewer: not addressed

### Steady-State Field Distributions in the Primary Loop

[ss_results] shows the steady-state scaled fission source and $^{135}Xe$ concentration distributions.

!media msr/msre/xe_poisoning/ss_results.png
       style=width:60%
       id=ss_results
       caption=Steady-state distributions in the MSRE loop: (a) scaled fission source and (b) $^{135}Xe$ concentration.

Key observations from the spatial fields include:

- The highest fission source is concentrated in the core region, with expected attenuation toward ex-core loop components.
- $^{135}Xe$ remains relatively low in most core regions because strong neutron flux drives Xe burnout to $^{136}Xe$ by absorption.
- A comparatively elevated $^{135}Xe$ band appears near the upper core / outlet region, where transport and production-decay competition produce localized accumulation.
- Ex-core regions retain higher Xe than the core until recirculation through the lower plenum returns salt to high-flux zones.

### Reactivity Impact of Xe Poisoning

> A reference case without Xe poisoning was also evaluated. Comparison of $k_{eff}$ between poisoned and unpoisoned coupled cases indicates an estimated reactivity penalty of approximately -410 pcm, which is in good agreement with trends from the SAM-Griffin coupled study.

Author: Indeed, i think it is probably due to the combination of both. the value is also close in a parallel SAM-Griffin modeling study

> A reference case without Xe poisoning was also evaluated. Comparison of $k_{eff}$ between poisoned and unpoisoned coupled cases indicates an estimated reactivity penalty of approximately -410 pcm, which is in good agreement with trends from the SAM-Griffin coupled study.

Reviewer: what is the penalty now with the fixes in the model

> # MSRE Multiphysics Xenon Poisoning Model

Reviewer: I cant find in this doco or the other model doco a table that gives me the MSRE operating parameters?

We d want to fit the pump rate to get a target mdot. 
And similarly if we have two pump terms (one vertical one horizontal) to get a target mdot in the bypass loop
